service/src/main/java/tds/session/web/endpoints/SessionController.java

```diff
@@ -48,11 +50,14 @@ public SessionController(SessionService sessionService) {
 
     @RequestMapping(value = "/{sessionId}/pause", method = RequestMethod.PUT, produces = MediaType.APPLICATION_JSON_VALUE)
     @ResponseBody
-    ResponseEntity<PauseSessionResponse> pause(@PathVariable final UUID sessionId, @RequestBody final String newStatus) {
-        final PauseSessionResponse response = sessionService.pause(sessionId, newStatus)
-            .orElseThrow(() -> new NotFoundException("Could not find session id %s", sessionId));
+    ResponseEntity<Response<PauseSessionResponse>> pause(@PathVariable final UUID sessionId, @RequestBody final PauseSessionRequest request) {
+        final Response<PauseSessionResponse> response = sessionService.pause(sessionId, request);
 
-        URI location = linkTo(methodOn(SessionController.class).findSessionById(response.getSessionId())).toUri();
+        if (response.getErrors().length > 0) {
```

Comment:
What if we add a convenience method to the `Response` object `hasErrors()` or `isInError()`.  I feel like this is going to be a common occurrence throughout the code.